sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/models/InputStreamBinaryData.java

```diff
@@ -162,6 +148,11 @@ public BinaryData toReplayableBinaryData() {
         return readAndBuffer(this.content.get(), length);
```

Comment:
back to competing consumers (which was a problem before this PR): `toReplayableBinaryData` here makes original data unusable, so we should not really create a new object

Reply:
With a case like this, if we have competing consumers, we already have issues given there may be interleaved reads on the backing `InputStream`. But, in cases where the backing content may only be consumable once, maybe the better option for buffering is to mutate the object itself rather than creating a new instance. No matter what, we'll still have the race condition issues on consumption but maybe this is a case where we need to just explicitly state that `BinaryData` may not be thread-safe and to consume appropriately.